### DISCUSSION TITLE
[WASM][AOT] Let's talk performance

### DISCUSSION THREAD

**Discussion by DierkDroth** [unoplatform/uno#8417] [Q&A]
I'm trying to understand where we stand performance wise with UNO WASM. E.g. I'm experiencing that my UNO WASM app (latest UNO 4.2 bits, .NET 6) running on a state-of-the-art iPad shows outright horrible runtime performance.
(yes, I understand that UNO iOS would be another option. However, that's not my point here)

Couple of questions:
- where do we stand regarding FullAOT and performance? I'm aware of the threads [here ](https://github.com/dotnet/runtime/issues/56309)and [here](https://github.com/unoplatform/uno/issues/4613). Are there any pending (WASM) issues which prevent FullAOT (aka forcing a fallback to interpreter) at this point or could I assume that what I see right now is the best (AOT) performance I will get (more or less)?
- are there guidelines available on how to profile a UNO WASM app to understand what could be optimized in my code?

Thanks in advance

**Comment by jeromelaban** [comment]
- Performance is being worked on for .NET 7 by the .NET team. You'll be able to test those bits soon in an Uno.Wasm.Bootstrap update.
- FullAOT is unlikely to be available any time soon and it will be disabled in the uno boostrapper, as there are still missing features in WebAssembly to enable it. The .NET team would be best to answer to question.
- If you want to profile an app, you can use the browser profiling tools.

**Reply by DierkDroth** [reply]
Thanks for follow up @jeromelaban. OK, I'll sit tight for a few more weeks to see how things will improve...
